Content.Server/VendingMachines/VendingMachineSystem.cs

```diff
@@ -0,0 +1,43 @@
+using System.Linq;
+using Content.Server.Items;
+using Content.Server.PowerCell.Components;
+using Content.Server.Stunnable.Components;
+using Content.Server.Weapon.Melee;
+using Content.Shared.ActionBlocker;
+using Content.Shared.Audio;
+using Content.Shared.Examine;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Notification.Managers;
+using Content.Shared.Throwing;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Content.Server.VendingMachines.Components;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using Content.Shared.DragDrop;
+using Content.Shared.VendingMachines;
+
+namespace Content.Server.VendingMachines
+{
+    public class VendingMachineSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SharedVendingMachineRestockComponent, DragDropRequestEvent>(OnDropRestockCompOnVendingMachine);         
+        }
+
+        private void OnDropRestockCompOnVendingMachine(EntityUid eUI, SharedVendingMachineRestockComponent restockComp, DragDropRequestEvent args)
```

Comment:
I tried using an entity system, but I couldn't make it handle drag drop events :/